File: Elearn-DataTier/src/main/java/via/dk/elearn/models/University.java
package via.dk.elearn.models;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import javax.persistence.*;

@Entity
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class University {
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Long id;

    @Column(unique = true, name = "university_name", length = 100)
    private String universityName;

    @Column(unique = true, name = "description", length = 1000)
    private String description;

    public University(String universityName, String description) {
        this.universityName = universityName;
        this.description = description;
    }
}

File: Elearn-ApplicationTier/Elearn.Shared/Models/University.cs
namespace Elearn.Shared.Models;

public class University
{
    public long Id { get; set; }
    public string Name { get; set; }
    
    public string Description { get; set; }
    public University()
    {
        
    }
    public University(long id, string name)
    {
        Id = id;
        Name = name;
    }

    
}